libobs/obs-encoder.c

```diff
@@ -325,11 +325,10 @@ static void add_connection(struct obs_encoder *encoder)
 	}
 
 	if (encoder->encoder_group) {
-		bool ready = false;
 		pthread_mutex_lock(&encoder->encoder_group->mutex);
-		encoder->encoder_group->encoders_started += 1;
-		ready = encoder->encoder_group->encoders_started ==
-			encoder->encoder_group->encoders_added;
+		encoder->encoder_group->num_encoders_started += 1;
+		bool ready = encoder->encoder_group->num_encoders_started >=
+			     encoder->encoder_group->num_encoders;
```

Comment:
I don't have a test harness to check removing encoders from groups that are being activated, but if I'm not misreading things you can get encoder groups "stuck" via the following steps:
1. have multiple encoders grouped
2. start only a few of the grouped encoders, wait for all started encoders to be added to `num_encoders_started` so that their `bool ready` check is done
3. remove non-started encoders
in that scenario the `ready` check will never happen again, and thus `add_ready_encoder_group` will never be called